### ISSUE TITLE
Refactor `ariaLabel` props to `aria-label`

### ISSUE THREAD

**Issue by ItsMarcoMSF** [carbon-design-system/carbon#12802]
<details>
  <summary>Click to expand details</summary>
  
### Package

carbon-components, carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

carbon-components@v10.48.0 & carbon-components-react@7.48.0

### React version

v16.14.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

</details>

 ### Description
 
 According to @carbon/react@1.17.0 [storybook](https://react.carbondesignsystem.com/?path=/story/getting-started-welcome--welcome), the Modal component accepts a prop called `aria-label`. 
 
 ![Screenshot 2022-12-06 at 11 06 59 AM](https://user-images.githubusercontent.com/56201575/205992241-28771f13-0e83-4b12-a36d-f02c648944cf.png)
 
 However, in most other Carbon components in react, the prop `ariaLabel` is accepted
 
 ![Screenshot 2022-12-06 at 11 29 13 AM](https://user-images.githubusercontent.com/56201575/205992553-b1d77f4d-489b-48a6-bad7-6681ade5a800.png)
 
 Can there be at least consistency between components in this area? A deeper dive into the source can also show that a number of components still accepts `aria-label` as a prop while the storybook says `ariaLabel`. And then there are other components that only accept one of `ariaLabel` or `aria-label` and not both. 
 
 According to React's official document on WAI-ARIA: https://reactjs.org/docs/accessibility.html#wai-aria:
 
 > Note that all aria-* HTML attributes are fully supported in JSX. Whereas most DOM properties and attributes in React are camelCased, these attributes should be hyphen-cased (also known as kebab-case, lisp-case, etc) as they are in plain HTML
 
So perhaps, the best approach is to support `aria-label` in all components?
 
 ### WCAG 2.1 Violation
 
_No response_
 
 ### Reproduction/example
 
 Not applicable.
 
 ### Steps to reproduce
 
 Not applicable.
 
 ### Code of Conduct
 
 - [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
 - [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

---

# Plan of action

[As described below](https://github.com/carbon-design-system/carbon/issues/12802#issuecomment-1344363498), Components with the prop `ariaLabel` need to be refactored to `aria-label`. We wanted to ship this as part of v11, https://github.com/carbon-design-system/carbon/issues/9535, but it fell out of scope. There are currently 13 usages that need to be refactored: https://carbon-react-explorer.vercel.app/props/ariaLabel

```[tasklist]
### Components to update
- [x] Refactor ariaLabel to aria-label in StructuredListWrapper
- [x] Refactor ariaLabel to aria-label in OverflowMenu
- [x] Refactor ariaLabel to aria-label in NumberInput
- [x] Refactor ariaLabel to aria-label in NotificationButton
- [x] Refactor ariaLabel to aria-label in FilterableMultiSelect
- [x] Refactor ariaLabel to aria-label in Dropdown
- [ ] Refactor ariaLabel to aria-label in TableSelectRow
- [x] Refactor ariaLabel to aria-label in InlineCheckbox
- [ ] Refactor ariaLabel to aria-label in TableSelectAll
- [ ] Refactor ariaLabel to aria-label in TableExpandRow
- [ ] Refactor ariaLabel to aria-label in TableExpandHeader
- [x] Refactor ariaLabel to aria-label in ComboBox
- [x] Refactor ariaLabel to aria-label in CodeSnippet
```

### Steps to complete for each component
- [ ] Deprecate `ariaLabel` by updating to use `deprecate()`
- [ ] Add `aria-label` prop -- e.g. `['aria-label']: PropTypes.string,`
- [ ] Refactor usages of `ariaLabel` to prefer `aria-label` if it's provided. Both props need to work for the time being to remain backwards compatibility

**Comment by tay1orjones** [comment]
Thanks for bringing this up @ItsMarcoMSF - yes, usages of `ariaLabel` should be converted to be `aria-label` instead.

We intended to ship this as part of v11, https://github.com/carbon-design-system/carbon/issues/9535, but it fell out of scope. There are currently 16 usages that need to be refactored: https://carbon-react-explorer.vercel.app/props/ariaLabel

Each component needs:
- [ ] Deprecate `ariaLabel` by updating to use `deprecate()`
- [ ] Add `aria-label`: `['aria-label']: PropTypes.string,`
- [ ] Refactor usages of `ariaLabel` to prefer `aria-label` if provided - both props need to work for the time being to remain backwards compatibility

This will fall into a lower priority category for our team, but we'll happily accept PRs if you're open to contributing!

**Comment by tay1orjones** [comment]
I just updated the issue title and body with a new "Plan of action" section outlining the work that needs done. 

This issue is a lower priority item for our team, but we'll happily accept PRs if you're open to contributing!

**Comment by tay1orjones** [comment]
@ItsMarcoMSF Whatever timeline works for you! We won't be able to pick this up for quite some time unfortunately.

It's all in the [contributing.md](https://github.com/carbon-design-system/carbon/blob/main/.github/CONTRIBUTING.md), but at a high level just fork the repo into your personal account, make a branch, make the changes, commit/push up, and open a PR. As this is a change to props on a component you may need to regenerate the public API snapshot via `yarn test -u` at the root after making your changes, but we can triage that in the pull request reviews. 

Let me know if I can help get you up and running. You could start with a single component and just ref this issue #

**Comment by jsehull** [comment]
Hey team - good find with the aria inconsistencies. I'm going to start working on some fixes for this.

**Comment by jsehull** [comment]
@tay1orjones this is a rather large body of work for all of this. I could break it down into multiple PRs for merging to get some things out there door sooner. How would you prefer something like this be handled?

**Comment by jsehull** [comment]
@tay1orjones Can you clarify what to do with the following regarding `aria-label`?
- `FilterableMultiSelect`: on the list, but does not have it 
- `NumberInput`: on the list, but does not have it 
- Notifications
- 1. `ActionableNotification` has `aria-label` but is not on Issue list above. Update or remove?
- 2. `ToastNotification` has the propType listed, but it is not currently used. Update or remove?

**Comment by tay1orjones** [comment]
@jsehull Yeah, I would sequence it out into separate PRs per component to get it out the door sooner.

The list of components to refactor was generated by searching the codebase for `ariaLabel: PropTypes`. Every usage listed there needs to be refactored to use `aria-label` unless the prop is not used (and not spread onto a child via `...other` or similar)

InlineCheckbox is used in a few spots - TableSelectRow, TableSelectAllRow, etc. and can be viewed in those DataTable stories using those components. 

Also I think we can just ignore any feature requests for now and focus on updating the existing API in the state it's in.

**Comment by jsehull** [comment]
Below are some of my progress updates in case others are monitoring.

**Items removed**:
1. NumberInput

**Completed items** (I'm running into issues getting PRs setup properly):
1. StructuredListWrapper (PR in progress)
2. OverflowMenu (PR in progress)
3. Dropdown
4. ComboBox
5. CodeSnippet

**In process:**
1. Notification (mostly done)
2. FilterableMultiSelect (mostly done)
3. all 4 table related items (half done)
- I get the new `aria-label` working but am having trouble keeping the `ariaLabel` functional

**Not started:**
1. Modal (not on original list, but still needs done)
2. InlineCheckbox

**Comment by jsehull** [comment]
8 of 12 completed and pending PR approvals.

4 of 12 remaining:
- TableSelectRow
- TableSelectAll
- TableExpandRow
- TableExpandHeader

The table section is proving more challenging, though the fix should be nearly the same. I'm having varying degrees of success getting those working.

At best, for some, I get the **new** `aria-label` working but am having trouble keeping the **deprecated** `ariaLabel` functional

___

For now I'm moving on to other Issues. Anyone is welcome to hop in and try these out!

Whoever works on this next can reference recent fixes as the updates should be similar.
___

Once a solution is found and properly tested, I run the following to ensure that snapshots are updated. It's common to have to fix tests afterwards. `DataTable` has many tests!

`yarn && yarn build && yarn test -u`

**Comment by TannerS** [comment]
@tay1orjones @tw15egan do yall mind if i help out with this?

**Comment by tw15egan** [comment]
Go for it @TannerS 🎉

**Comment by fitrahfm** [comment]
@tay1orjones @tw15egan May I help by working on this?

**Comment by tw15egan** [comment]
Sure thing @fitrahfm 👍🏻
